xbmc/utils/GpuInfo.h

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "utils/Temperature.h"
+
+#include <memory>
+#include <string>
+
+/*! \brief Class to concentrate all methods related to GPU information
+* \details This is used by the Info interface to obtain the current GPU temperature
+*/
+class CGPUInfo
+{
+public:
+  /*! \brief Getter from the specific platform GPUInfo
+   \return the platform specific implementation of GPUInfo
+   */
+  static std::shared_ptr<CGPUInfo> GetGPUInfo();
+
+  /*! \brief Get the temperature of the GPU
+   \param[in,out] temperature - the temperature to fill with the result
+   \return true if it was possible to obtain the GPU temperature, false otherwise
+   */
+  bool GetTemperature(CTemperature& temperature) const;
```

Comment:
Can we change this to return `std::optional<CTemperature>` instead of returning book and using an in/out param?

Reply:
Sure

Reply:
Actually I'm not sure if this is a good idea...likely it's better as is. `CTemperature` has a `isValid` method and the calling API stores the gpu temperature as a member and only requests it from time to time (while returning the stored temperature on the info interface)
We need to return `CTemperature` and set it invalid for presentation on the GUI if we could not obtain a temperature value.

Reply:
Long story short, this can be done but better as a follow up since it requires changes to pretty much all uses of CTemperature.